File: app/src/main/java/com/biryanistudio/todo/Db/TasksDbHelper.java
package com.biryanistudio.todo.db;

import android.content.Context;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;
import android.support.annotation.NonNull;

public class TasksDbHelper extends SQLiteOpenHelper {
    private static final int DATABASE_VERSION = 1;
    private static final String DATABASE_NAME = "Tasks.db";
    private static TasksDbHelper dbHelper;

    private TasksDbHelper(Context context) {
        // If you change the database schema, you must increment the database version.
        super(context, DATABASE_NAME, null, DATABASE_VERSION);
    }

    public static TasksDbHelper getInstance(@NonNull Context context) {
        if (dbHelper == null) {
            dbHelper = new TasksDbHelper(context);
            return dbHelper;
        } else {
            return dbHelper;
        }
    }

    //TODO: We are not calling this method anywhere.
    public static void closeDbHelper() {
        dbHelper.getReadableDatabase().close();
        dbHelper.close();
    }

    public void onCreate(SQLiteDatabase db) {
        db.execSQL(TasksContract.SQL_CREATE_ENTRIES);
    }

    public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
        // This database is only a cache for online data, so its upgrade policy is
        // to simply to discard the data and start over
        db.execSQL(TasksContract.SQL_DELETE_ENTRIES);
        onCreate(db);
    }

    public void onDowngrade(SQLiteDatabase db, int oldVersion, int newVersion) {
        onUpgrade(db, oldVersion, newVersion);
    }
}

File: app/src/main/java/com/biryanistudio/todo/database/TasksDbHelper.kt
package com.biryanistudio.todo.database

import android.content.Context
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteOpenHelper

internal class TasksDbHelper private constructor(context: Context) : SQLiteOpenHelper(context,
        TasksDbHelper.DATABASE_NAME, null, TasksDbHelper.DATABASE_VERSION) {

    override fun onCreate(db: SQLiteDatabase) {
        db.execSQL(TasksContract.SQL_CREATE_ENTRIES)
    }

    override fun onUpgrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) {
        db.execSQL(TasksContract.SQL_DELETE_ENTRIES)
        onCreate(db)
    }

    override fun onDowngrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) {
        onUpgrade(db, oldVersion, newVersion)
    }

    companion object {
        private val DATABASE_VERSION = 1
        private val DATABASE_NAME = "Tasks.db"
        private var dbHelper: TasksDbHelper? = null

        fun getInstance(context: Context): TasksDbHelper {
            if (dbHelper == null) dbHelper = TasksDbHelper(context)
            return dbHelper as TasksDbHelper
        }
    }
}